src/Control/Monad/Trace.hs

```diff
@@ -163,7 +163,7 @@ instance (MonadBaseControl IO m, MonadIO m) => MonadTrace (TraceT m) where
                   modifyTVar' (tracerPendingCount tracer) (\n -> n - 1)
                   tags <- readTVar tagsTV
                   logs <- sortOn (\(t, k, _) -> (t, k)) <$> readTVar logsTV
-                  writeTChan (tracerChannel tracer) (Sample spn tags logs start (end - start))
+                  writeTBQueue (tracerQueue tracer) (Sample spn tags logs start (end - start))
```

Comment:
`writeTBQueue` blocks when the queue is full, it should instead do nothing (see [SBQueue](https://github.com/scrive/log/blob/6bf80cab8d9684a6c1dfde3f35edbdff40bd1c40/log-base/src/Log/Logger.hs#L127), in particular [writeSBQueue](https://github.com/scrive/log/blob/6bf80cab8d9684a6c1dfde3f35edbdff40bd1c40/log-base/src/Log/Logger.hs#L152)).